packages/authentication/src/loaders/providers.ts

```diff
@@ -0,0 +1,49 @@
+import { LoaderOptions, ModulesSdkTypes } from "@medusajs/types"
+import { asClass } from "awilix"
+import * as defaultProviders from "@providers"
+import { AuthProviderService } from "@services"
+import { ServiceTypes } from "@types"
+
+export default async ({
+  container,
+  options,
+}: LoaderOptions<
+  | ModulesSdkTypes.ModuleServiceInitializeOptions
+  | ModulesSdkTypes.ModuleServiceInitializeCustomDataLayerOptions
+>): Promise<void> => {
+  // if(options.providers?.length) {
+  // TODO: implement plugin provider registration
+  // }
+
+  const providersToLoad = Object.values(defaultProviders)
+
+  const authProviderService: AuthProviderService =
+    container.cradle["authProviderService"]
+
+  const providers = await authProviderService.list({
+    provider: providersToLoad.map((p) => p.PROVIDER),
+  })
+
+  const loadedProviders = new Map(providers.map((p) => [p.provider, p]))
+
+  const providersToCreate: ServiceTypes.CreateAuthProviderDTO[] = []
+
+  for (const provider of providersToLoad) {
+    container.registerAdd("providers", asClass(provider).singleton())
```

Comment:
suggestion: could we not use registerAdd and instead create a providers array that we register after the loop?